src/mem/chunkmap.h

```diff
@@ -47,14 +47,19 @@ namespace snmalloc
 // Use flat map is under a single node.
 #  define SNMALLOC_MAX_FLATPAGEMAP_SIZE PAGEMAP_NODE_SIZE
 #endif
-  static constexpr bool USE_FLATPAGEMAP = pal_supports<LazyCommit>() ||
-    (SNMALLOC_MAX_FLATPAGEMAP_SIZE >=
-     sizeof(FlatPagemap<SUPERSLAB_BITS, uint8_t>));
 
-  using ChunkmapPagemap = std::conditional_t<
-    USE_FLATPAGEMAP,
-    FlatPagemap<SUPERSLAB_BITS, uint8_t>,
-    Pagemap<SUPERSLAB_BITS, uint8_t, 0>>;
+  template<typename T>
+  static constexpr bool use_flatpagemap_for()
```

Comment:
For what?  The name implies that the function takes an argument, but it does not.

I had to read the implementation to discover that it means 'use flatmap for `T`'.  

Please add a doc comment.

Reply:
Is a C++14 templatized variable declaration OK rather than a function?

Reply:
Happy with C++14, I think the code requires C++17 in some places.